CorePayments/src/main/java/com/paypal/android/corepayments/graphql/common/GraphQLClient.kt

```diff
@@ -28,9 +32,11 @@ internal class GraphQLClient(
         "Origin" to coreConfig.environment.graphQLEndpoint
     )
 
-    suspend fun send(graphQLRequestBody: JSONObject): GraphQLResponse {
+    suspend fun send(graphQLRequestBody: JSONObject, queryName: String? = null): GraphQLResponse {
         val body = graphQLRequestBody.toString()
-        val httpRequest = HttpRequest(graphQLURL, HttpMethod.POST, body, httpRequestHeaders)
+        val urlString = if (queryName != null) "$graphQLURL?$queryName" else graphQLURL
+        val httpRequest = HttpRequest(URL(urlString), HttpMethod.POST, body, httpRequestHeaders)
+
         val httpResponse = http.send(httpRequest)
         val correlationId: String? = httpResponse.headers[PAYPAL_DEBUG_ID]
         val status = httpResponse.status
```

Comment:
It doesn't look like we're parsing the errors in the graphQL response anywhere below?

Reply:
I was just messing around with this API and on an error case I get:

```
"{\"error\":\"Unauthorized.  Missing a required scope.  Retry your request with a valid access token or session token.\",\"correlation_id\":\"f906092501981\",\"sys\":{\"links\":{\"jsBaseUrl\":\"https://www.sandbox.paypal.com/js\",\"cssBaseUrl\":\"https://www.sandbox.paypal.com/css\",\"templateBaseUrl\":\"https://www.sandbox.paypal.com/templates/GB/en\",\"resourceBaseUrl\":\"https://www.sandbox.paypal.com\",\"originalTemplateBaseUrl\":\"https://www.sandbox.paypal.com/templates\"},\"pageInfo\":{\"date\":\"Aug 11, 2023 08:58:09 -07:00\",\"hostName\":\"rZJvnqaaQhLn/nmWT8cSUjOx898qoYZ0Tm9DnErguV/rx24bGoErTLFawZbGZPF+\",\"rlogId\":\"rZJvnqaaQhLn%2FnmWT8cSUnQp6l6UCEzKKo4FJOfpRh5MUWrZYnpzB1v%2BWKb1idf2c4FoJZOAZVPglD9xAmGZsQ_189e551c697\",\"script\":\"node\",\"debug\":{\"version\":{\"manifestId\":\"1.0.3_20230719191007168\"}}},\"locality\":{\"timezone\":{\"determiner\":\"viaCowPrimary\",\"value\":\"Europe/London\"},\"country\":\"GB\",\"locale\":\"en_GB\",\"language\":\"en\",\"directionality\":\"ltr\"},\"tracking\":{\"fpti\":{\"name\":\"pta\",\"jsURL\":\"https://www.paypalobjects.com\",\"serverURL\":\"https://t.paypal.com/ts\",\"dataString\":\"pgrp=graphqlnodeweb%2F.dust&page=graphqlnodeweb%2F.dust&pgst=1691769489047&calc=f906092501981&nsid=a9CV3X3D2PUZsDN7cifk1v4Xg8Cprsr4&pgtf=Nodejs&env=sandbox&s=ci&csci=26b13e4a47ef4e338219454b9e10db83&comp=graphqlnodeweb&tsrce=graphqlnodeweb&cu=0&ef_policy=gdpr_v2.1\"}}}}"
```

Looks like currently the details under `error` wouldn't be surface by our SDK (same issue on iOS too)

Reply:
> It doesn't look like we're parsing the errors in the graphQL response anywhere below?

Yeah I don't think we every got to the point of formalized error handling for GraphQL. It's something we need to look into. It looks like they are standardized in the [GraphQL spec](https://spec.graphql.org/October2021/#sec-Errors) at least, but we may want to go a step further if we can provide more meaningful error messages to merchants.

Reply:
I'm surprised about the missing access token portion though. I thought all we needed was to provide `clientID` to the GraphQL request, but I wonder if there's more we need to do for auth.

Reply:
Oh the missing access token error I got was triggered by me just messing around to see what the error format looks like, not by using your from the demo app!

Reply:
Do we want to add the graphQL error handling to this PR, or can we make a JIRA so we don't forget?

Reply:
For GraphQL Error Handling: DTPPCPSDK-708